awesome_owl/static/src/counter/counter.js

```diff
@@ -0,0 +1,24 @@
+import { Component, useState } from "@odoo/owl";
+
+
+export class Counter extends Component {
+    static template = "awesome_owl.counter";
+    static defaultProps = {
+        buttonText: "Increment",
+    }
+    static props = {
+        buttonText: { type: String, optional: true, },
+        onChange: { type: Function, optional: true },
+    };
+
+    setup() {
+        this.state = useState({ value: 1 });
+    }
+
+    increment() {
+        this.state.value++;
+        if (this.props.onChange) {
+            this.props.onChange();
+        }
```

Comment:
just for your information, it's maybe less readable in this case but in JS you can do something like this


```suggestion
        this.props.onChange?.();
```